orcid-core/src/main/java/org/orcid/core/manager/impl/StatisticsCacheManagerImpl.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * =============================================================================
+ *
+ * ORCID (R) Open Source
+ * http://orcid.org
+ *
+ * Copyright (c) 2012-2014 ORCID, Inc.
+ * Licensed under an MIT-Style License (MIT)
+ * http://orcid.org/open-source-license
+ *
+ * This copyright and license information (including a link to the full license)
+ * shall be included in its entirety in all copies or substantial portion of
+ * the software.
+ *
+ * =============================================================================
+ */
+package org.orcid.core.manager.impl;
+
+import java.text.NumberFormat;
+import java.util.Locale;
+
+import javax.annotation.Resource;
+
+import net.sf.ehcache.Cache;
+import net.sf.ehcache.Element;
+
+import org.orcid.core.manager.StatisticsManager;
+import org.orcid.core.utils.statistics.StatisticsEnum;
+import org.orcid.jaxb.model.statistics.StatisticsSummary;
+import org.orcid.jaxb.model.statistics.StatisticsTimeline;
+import org.orcid.persistence.jpa.entities.StatisticKeyEntity;
+
+/**
+ * @author Shobhit Tyagi
+ */
+public class StatisticsCacheManagerImpl implements StatisticsCacheManager {
+
+    @Resource
+    StatisticsManager statisticsManager;
+
+    @Resource(name = "statisticsCache")
+    private Cache statisticsCache;
+    
+    LockerObjectsManager lockers = new LockerObjectsManager();
+
+	@Override
+	public StatisticsSummary retrieve() {
+        StatisticsSummary statisticsSummary = toStatisticsSummary(statisticsCache.get("statisticsValues"));
+        StatisticKeyEntity statisticKeyEntity = toStatisticsKey(statisticsCache.get("statisticsLatestKey"));
+        StatisticKeyEntity statisticKeyEntityLatest = statisticsManager.getLatestKey();
+        if(needsFresh(statisticsSummary, statisticKeyEntity, statisticKeyEntityLatest)) {
+        	try {
+                synchronized (lockers.obtainLock("Statistics")) {
```

Comment:
@rcpeters unlike the profile table, which locks a specific row(using orcid value), a table lock is required here. So I used the string "Statistics" as the key.